spec/index.html

```diff
@@ -186,9 +186,11 @@ <h3>Triple Terms</h3>
       For a <a>triple term</a> to be asserted,
       it must also appear in a graph as an <a>asserted triple</a>.</p>
 
-    <p>A <a>triple term</a> can be used as the object of a <a>triple</a> with the predicate <code>rdf:reifies</code>.
+    <p>A <a>triple term</a> can be used as the object of a <a>triple</a> with the predicate <code>rdf:reifies</code>,
+      called a <dfn>reified triple</dfn>.
```

Comment:
It is not clear from this text what exactly it is that is called a reified triple. Is it any triple term that is used as the object of a triple with the predicate `rdf:reifies`? Or is it any triple term, no matter where it is used? Or is it a triple with the predicate `rdf:reifies` and a triple term in its object position?

Reply:
The sentence begins "A triple term can be used as the object of a triple with the predicate `rdf:reifies`. It is such a triple which is called a "reified triple". Maybe the following makes it more clear:

```suggestion
    <p>A <a>triple term</a> can be used as the object of a <a>triple</a> with the predicate <code>rdf:reifies</code>,
      such a triple is called a <dfn>reified triple</dfn>.
```

Reply:
I am also struggling with this. If what I think it's saying *is* what it's saying, I think we've got to change some things.

Is this meant to say that the "reified triple" is the triple that could be inserted if the triple term were deconstructed, or that the "reified triple" is the triple that has `rdf:reifies` for a predicate?

Reply:
I think it pretty plainly says that a triple with predicate `rdf:reifies` and a triple term object is a "reified triple". This is consistent with the `reifiedTriple` production in Turtle, and is how we encourage people to actually use triple terms. If you have a suggested improvement to the wording, then please suggest away.

Reply:
@gkellogg thanks for the clarification about the target of the definition! In this case, calling such a triple (that has the predicate `rdf:reifies` and a triple term object) a "reified triple" does not sound right. More specifically, I wouldn't use the adjective "reified" for such a triple because that triple is not reified. Instead, that triple establishes/expresses the reification. Therefore, calling it "reifying triple" or "reification triple" sounds better to me.

Reply:
We've been using the term "reified triple" for a while, but I'm happy to bikeshed it. The Turtle grammar rule is [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple), so we'll want to coordinate updating terminology.

> A [reified triple](https://www.w3.org/TR/rdf12-concepts/#dfn-reified-triple) is represented using the [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple) production starting with [<<](https://www.w3.org/TR/rdf12-turtle/#cp-double-lt), followed by either a [subject](https://www.w3.org/TR/rdf12-turtle/#grammar-production-subject) or [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple), a [predicate](https://www.w3.org/TR/rdf12-turtle/#grammar-production-predicate), and a [ttObject](https://www.w3.org/TR/rdf12-turtle/#grammar-production-ttObject), followed by an optional [reifier](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifier), composed of a [~](https://www.w3.org/TR/rdf12-turtle/#cp-tilde) followed by an optional [iri](https://www.w3.org/TR/rdf12-turtle/#grammar-production-iri) production or [BlankNode](https://www.w3.org/TR/rdf12-turtle/#grammar-production-BlankNode) production, and ending with [>>](https://www.w3.org/TR/rdf12-turtle/#cp-double-gt). For example, << :subject :predicate :object ~ :IRIREF >>. If no reifiers are present, or a [reifier](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifier) is not immediately followed by an [iri](https://www.w3.org/TR/rdf12-turtle/#grammar-production-iri) or [BlankNode](https://www.w3.org/TR/rdf12-turtle/#grammar-production-BlankNode), a fresh [RDF blank node](https://www.w3.org/TR/rdf12-concepts/#dfn-blank-node) is allocated, as with << :subject :predicate :object >>, or << :subject :predicate :object ~ >>.

`reifiedTriple` would have been short for `reifiedTripleTerm`, as the construct that relates a "reifier" with a "triple term". We should probably discuss on a call to get consensus on such terminology.

Reply:
@gkellogg now you are confusing me completely :-/

According to [your earlier comment above](https://github.com/w3c/rdf-concepts/pull/98#discussion_r1714483285), "a triple with predicate `rdf:reifies` and a triple term object is a "reified triple"." So, I thought I had understood that the notion you are calling a "reified triple" is every RDF triple (_s_, _p_, _o_) such that _p_ is the IRI `rdf:reifies` and _o_ is a triple term.

Yet, now you have merged @TallTed's [recent edit suggestion](https://github.com/w3c/rdf-concepts/pull/98#discussion_r1715805409), according to which the word "reified triple" is meant to refer to a triple term _o_ if _o_ is the object of an RDF triple (_s_, _p_, _o_) with _p_=`rdf:reifies`. So, it is just the triple term that is called "reified triple"?

Your [previous comment](https://github.com/w3c/rdf-concepts/pull/98#discussion_r1715572839) with the quote from [Sec.2.10 of the new version of the Turtle doc](https://www.w3.org/TR/rdf12-turtle/#reified-triples) increases the confusion by introducing yet another possible (and different!) meaning of the word "reified triple". Here, the word "reified triple" is introduced to refer to a particular form of notation/subexpressions to be used (at specific positions) within a Turtle file; namely, the subexpressions that are accepted by the [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple) production. If that is indeed the intended meaning of the word "reified triple", then I wonder why this notion needs to be introduced in RDF Concepts, as it is something specific to a particular serialization format (or perhaps to a class of such formats). In this case, I also notice that the very first sentence of the aforementioned [Sec.2.10](https://www.w3.org/TR/rdf12-turtle/#reified-triples) of TURTLE-12 is completely wrong in itself.

Which of these three concepts is it that you are calling a "reified triple".
1. Is it an RDF triple  (_s_, _p_, _o_) for which it holds that _p_ is the IRI `rdf:reifies` and _o_ is a triple term?
2. Or is it a triple term that occurs as the object of an RDF triple that has `rdf:reifies` as it predicate?
3. Or is it an expression that is accepted by the [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple) production?

Reply:
In my mind, 

* The following are all reifications of the same reified triple, identifying it as `:iri` in the last, and as a blank node in the other three:
  * `<< :subject :predicate :object >>`
  * `<< :subject :predicate :object ~ >>`
  * `<< :subject :predicate :object ~ _:bn1 >>`
  * `<< :subject :predicate :object ~ :iri >>` 

  The first two omit an identifier, and the latter two include an explicit identifier. All of these may appear in subject or object position (in at least some serializations); however, only the identifier gets written as a subject to the graph, always with a predicate of rdf:reifies and an object of the first form written in this bullet.

* `:a rdf:reifies << :subject :predicate :object >>` is a reifying triple.

That seems to me to be what is said in [the quoted block](https://github.com/w3c/rdf-concepts/pull/98#discussion_r1715572839) above, but it is not what is said by —
```
    <p>A <a>triple term</a> can be used as the object of a <a>triple</a> with the predicate <code>rdf:reifies</code>,
      called a <dfn>reified triple</dfn>.
```
```suggestion
    <p>A <a>triple term</a> can be used as the object of a <a>triple</a> with the predicate <code>rdf:reifies</code>,
      upon which use the <a>triple term</a> may be called a <dfn>reified triple</dfn>.
```
I think the quoted block above must change, mostly as below, with the problem that `reifier` is now triply defined as 
* `~ {iri|blanknode|''}`, i.e., including the tilde — when it's used within the `<< >>` construct
* `{iri|blanknode|''}` — when it's used as an identifier in subject or object position
* `[~](https://www.w3.org/TR/rdf12-turtle/#cp-tilde)` — when it's not `immediately followed by an iri production or a BlankNode production`

I think the `~` must become a `reifier delimiter` when it's used to separate an object from a `reifier` (which must be either a `uri production` or `BlankNode production`), something like what I've tweaked the block to say:

> A [reified triple](https://www.w3.org/TR/rdf12-concepts/#dfn-reified-triple) is represented using the [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple) production starting with [<<](https://www.w3.org/TR/rdf12-turtle/#cp-double-lt), followed by a [subject](https://www.w3.org/TR/rdf12-turtle/#grammar-production-subject), a [predicate](https://www.w3.org/TR/rdf12-turtle/#grammar-production-predicate), and a [ttObject](https://www.w3.org/TR/rdf12-turtle/#grammar-production-ttObject), followed by an optional [reifier](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifier), composed of a [~](https://www.w3.org/TR/rdf12-turtle/#cp-tilde) followed by an optional [iri](https://www.w3.org/TR/rdf12-turtle/#grammar-production-iri) production or [BlankNode](https://www.w3.org/TR/rdf12-turtle/#grammar-production-BlankNode) production, and ending with [>>](https://www.w3.org/TR/rdf12-turtle/#cp-double-gt). For example, << :subject :predicate :object ~ :IRIREF >>. If no reifiers are present, or a [reifier](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifier) delimiter (tilde, `~`) is not immediately followed by an [iri](https://www.w3.org/TR/rdf12-turtle/#grammar-production-iri) or [BlankNode](https://www.w3.org/TR/rdf12-turtle/#grammar-production-BlankNode), a fresh [RDF blank node](https://www.w3.org/TR/rdf12-concepts/#dfn-blank-node) is allocated, as with << :subject :predicate :object >> or << :subject :predicate :object ~ >>.

(I've taken a couple days to write this, and I'm not sure I carried all my thoughts through to completion. Please ask, if anything seems unclear.)

Reply:
@hartig wrote:

> Yet, now you have merged @TallTed's [recent edit suggestion](https://github.com/w3c/rdf-concepts/pull/98#discussion_r1715805409), according to which the word "reified triple" is meant to refer to a triple term _o_ if _o_ is the object of an RDF triple (_s_, _p_, _o_) with _p_=`rdf:reifies`. So, it is just the triple term that is called "reified triple"?

Yes, that merge was too hasty, and is incorrect.

> Your [previous comment](https://github.com/w3c/rdf-concepts/pull/98#discussion_r1715572839) with the quote from [Sec.2.10 of the new version of the Turtle doc](https://www.w3.org/TR/rdf12-turtle/#reified-triples) increases the confusion by introducing yet another possible (and different!) meaning of the word "reified triple". Here, the word "reified triple" is introduced to refer to a particular form of notation/subexpressions to be used (at specific positions) within a Turtle file; namely, the subexpressions that are accepted by the [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple) production. If that is indeed the intended meaning of the word "reified triple", then I wonder why this notion needs to be introduced in RDF Concepts, as it is something specific to a particular serialization format (or perhaps to a class of such formats). In this case, I also notice that the very first sentence of the aforementioned [Sec.2.10](https://www.w3.org/TR/rdf12-turtle/#reified-triples) of TURTLE-12 is completely wrong in itself.

I expect most concrete syntaxes (RDF and SPARQL) to support a construct such as Turtle's [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple). I think it's an important concept, as is _reifier_, that may not have a direct impact on the abstract syntax, but is never-the-less an important RDF-star concept. We expect people to use reified triples to write statements about triples rather than using triple terms directly.

> Which of these three concepts is it that you are calling a "reified triple".
> 
> 1. Is it an RDF triple  (_s_, _p_, _o_) for which it holds that _p_ is the IRI `rdf:reifies` and _o_ is a triple term?
> 2. Or is it a triple term that occurs as the object of an RDF triple that has `rdf:reifies` as it predicate?
> 3. Or is it an expression that is accepted by the [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple) production?

For RDF Concepts, it should be the first. The [reifiedTriple](https://www.w3.org/TR/rdf12-turtle/#grammar-production-reifiedTriple) production is Turtle's way of expressing that in syntax. Of course, the WG may ultimately decide to broaden this to `rdf:reifies` or a sub-property.


@TallTed wrote:

> In my mind,
> 
> * The following are all reifications of the same reified triple, identifying it as `:iri` in the last, and as a blank node in the other three:
>   
>   * `<< :subject :predicate :object >>`
>   * `<< :subject :predicate :object ~ >>`
>   * `<< :subject :predicate :object ~ _:bn1 >>`
>   * `<< :subject :predicate :object ~ :iri >>`
>   
>   The first two omit an identifier, and the latter two include an explicit identifier. All of these may appear in subject or object position (in at least some serializations); however, only the identifier gets written as a subject to the graph, always with a predicate of rdf:reifies and an object of the first form written in this bullet.

Yes, that is correct for Turtle.

> * `:a rdf:reifies << :subject :predicate :object >>` is a reifying triple.

Not quite, as a triple term is written with `<<( )>>`, so it would rather be:

* `:a rdf:reifies <<( :subject :predicate :object )>>` is a reifying triple.

Your suggested change above is not what I had in mind. My intention is that a "reified triple" is a triple using `rdf:reifies` where the object is a "triple term". It is not that triple term which is a reified triple.

> I think the quoted block above must change, mostly as below, with the problem that `reifier` is now triply defined as
> 
> * `~ {iri|blanknode|''}`, i.e., including the tilde — when it's used within the `<< >>` construct
> * `{iri|blanknode|''}` — when it's used as an identifier in subject or object position
> * `[~](https://www.w3.org/TR/rdf12-turtle/#cp-tilde)` — when it's not `immediately followed by an iri production or a BlankNode production`

These are syntactic constructs, much as an IRIREF in Turtle uses `<` and `>`, but is not part of the IRI itself, the `reifier` production defines a reifier for which the tilde is a syntactic component.

> I think the `~` must become a `reifier delimiter` when it's used to separate an object from a `reifier` (which must be either a `uri production` or `BlankNode production`), something like what I've tweaked the block to say:

The really belongs in the Turtle repo, not Concepts, but I don't see why we would explicitly name tilde as a reifier delimiter any more than we would name `<` and `>` as IRI delimiters.